R/ignore_step.R

```diff
@@ -0,0 +1,71 @@
+#' Remove steps from recipe
+#'
+#' `ignore_step` will return a recipe without steps specified by the `number` or
+#' `id` argument.
+#' 
+#' @param x A `recipe` object.
+#' @param number An integer vector, Denoting the positions of the steps that 
+#'   should be removed.
+#' @param id A character string. Denoting the `id` of the steps that should be
+#'   removed.
+#' 
+#' @details
+#' `number` or `id` must be specified. Specifying neither or both will result
+#' in a error.
+#' 
+#' @return a `recipe` object.
+#'
+#' @examples
+#' rec <- recipe(mpg ~ ., data = mtcars) %>%
+#'   step_dummy(all_nominal_predictors()) %>%
+#'   step_impute_mean(all_numeric_predictors()) %>%
+#'   step_normalize(all_numeric_predictors()) %>%
+#'   step_pca(all_numeric_predictors(), id = "PCA")
+#' 
+#' ignore_step(rec, number = 1)
+#' 
+#' ignore_step(rec, number = 1:2)
+#' 
+#' ignore_step(rec, id = "PCA")
+#' @export
+ignore_step <- function(x, number, id) {
```

Comment:
I would give `number` and `id` some default values since only one of the two is required. `tidy()` uses `NA`.